pages/docs/coding-conventions/testing.en-US.mdx

```diff
@@ -0,0 +1,108 @@
+# Testing
+
+We use a combination of unit testing, integration testing, and e2e testing to ensure the reliability and maintainability of O3. We use the following tools:
+
+- [Jest](https://vitest.dev) and [React Testing Library](https://testing-library.com/react) for unit and integration testing.
+- [Playwright](https://playwright.dev) for e2e testing.
+
+The following are some general guidelines for testing:
+
+### Unit and integration testing
+
+- Avoid these [common testing pitfalls](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library):
```

Comment:
This section is confusing. Some points (like "Write fewer, longer tests") I have to re-read to understand whether I should be avoiding or not.

Reply:
Yeah, I agree Kent's blog title there is potentially misleading. It does make a lot of sense, though, and the main premise is:

> Don't worry about having long tests. When you're thinking about your two users and [avoid the test user](https://kentcdodds.com/blog/avoid-the-test-user), then your tests will often involve multiple assertions and that's a good thing. Don't arbitrarily separate your assertions into individual test blocks, there's no good reason to do so.

So maybe I could rephrase that better to say something like: "Write fewer, more comprehensive tests that cover complete scenarios, rather than many small isolated tests"

Reply:
I meant making some of the things to avoid more explicit, because some of the are "do"s and some are "don't"s.

Reply:
Hopefully things are a lot more clearer in my more recent commits.